README.md

```diff
@@ -333,25 +333,106 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   ```swift
   // WRONG
-  let host: Host = Host()
+  let sun: Star = Star(mass: 1.989e30)
+  let earth: Planet = Planet.earth
 
   // RIGHT
-  let host = Host()
+  let sun = Star(mass: 1.989e30)
+  let earth = Planet.earth
+
+  // WRONG: Most literals provide a default type that can be inferred.
+  let enableGravity: Bool = true
+  let numberOfPlanets: Int = 8
```

Comment:
@calda how do we handle/guide on the case where you want to tell the language that a literal should be interpreted in a special way? For example, one way to get a floating-point value of 100 is to write

```swift
let myFloatingPointValue: Double = 100
```

which will produce a different result than

```swift
let myFloatingPointValue = 100 // implicitly of type `Int`
```

In this example, another option is to write
```swift
let myFloatingPointValue = 100.0 // implicitly of type `Double`
```

but even then it's necessary to write an explicit type in some cases, like if you explicitly wanted a `Float` instead of `Double`
```swift
let myFloatingPointValue: Float = 100
```

Reply:
I am curious how our autocorrection handles this case where removing the type can change the meaning of the code, and I also think that situations like this may warrant some discussion in the prose of this rule.

Reply:
In my testing I see that the linter is smart enough to correct `let numberOfPlanets: Int = 8` to `let numberOfPlanets = 8` but not correct `let numberOfPlanets: Float = 8`. Wow! I wonder if you happen to have a code pointer to how this is implemented.

Reply:
Btw I also saw as I continued to review that this is covered in more detail in the next rule, though I wonder if we should move some of the content from the next rule up to this one since this rule is the one focused on removing types.

Reply:
I implemented this back in https://github.com/nicklockwood/SwiftFormat/pull/921! Here's the code: https://github.com/nicklockwood/SwiftFormat/pull/921/files#diff-1f5cbb52e333eb61c8d6fef2bd765e3e8e6fa8f3101f91f98a9dfce9e1ef480dR1363

Reply:
> I also saw as I continued to review that this is covered in more detail in the next rule, though I wonder if we should move some of the content from the next rule up to this one since this rule is the one focused on removing types.

Good idea. I think the next rule probably goes in to too much detail about literals, and that content would be a better fit for the discussion about the `redundantType` rule.